specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/networkVirtualAppliance.json

```diff
@@ -0,0 +1,509 @@
+        },
+        "x-ms-examples": {
+          "List all Network Virtual Appliances for a given subscription": {
+            "$ref": "./examples/NetworkVirtualApplianceListBySubscription.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "NetworkVirtualAppliance": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/NetworkVirtualAppliancePropertiesFormat",
+          "description": "Properties of the Network Virtual Appliance"
+        },
+        "virtualHub": {
+          "readOnly": false,
+          "$ref": "./network.json#/definitions/SubResource",
+          "description": "The Virtual Hub where Network Virtual Appliance is being deployed"
+        },
+        "etag": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets a unique read-only string that changes whenever the resource is updated"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/Resource"
+        }
+      ],
+      "description": "NetworkVirtualAppliance Resource."
+    },
+    "NetworkVirtualAppliancePropertiesFormat": {
+      "properties": {
+        "bootStrapConfigurationBlob": {
+          "type": "array",
+          "readOnly": false,
+          "description": "BootStrapConfigurationBlob storage URLs",
+          "items": {
+            "type": "string"
+          }
+        },
+        "cloudInitConfigurationBlob": {
+          "type": "array",
+          "readOnly": false,
+          "description": "CloudInitConfigurationBlob storage URLs",
+          "items": {
+            "type": "string"
+          }
+        },
+        "identity": {
+          "$ref": "./network.json#/definitions/ManagedServiceIdentity",
+          "description": "The service principal that has read access to cloud-init and config blob"
+        },
+        "virtualApplianceAsn": {
+          "type": "integer",
+          "readOnly": false,
+          "format": "int64",
+          "minimum": 0,
+          "maximum": 4294967295,
+          "description": "VirtualAppliance ASN"
+        },
+        "virtualApplianceNics": {
+          "type": "array",
+          "readOnly": true,
+          "description": "List of Virtual Appliance Network Interfaces",
+          "items": {
+            "$ref": "#/definitions/VirtualApplianceNicProperties"
+          }
+        },
+        "routeServiceAsn": {
+          "type": "integer",
+          "readOnly": true,
+          "format": "int64",
+          "minimum": 0,
+          "maximum": 4294967295,
+          "description": "RouteService ASN"
+        },
+        "routeServiceIps": {
+          "type": "array",
+          "readOnly": true,
+          "description": "RouteService Peering IP Addresses",
+          "items": {
+            "type": "string"
+          }
+        },
+        "Sku": {
```

Comment:
The sku property should be top level and follow this contract:
https://armwiki.azurewebsites.net/api_contracts/SKUUpdateContract%E2%80%8B%E2%80%8B.html

Reply:
I moved this to top level. But would need your help to change Sku as per ARM guidelines. I'm confused if I need all the properties listed by ARM guidelines here. Most of these would not be applicable to our service. Perhaps change name from Sku to nvaSku ?